### PR TITLE
New Blueprint Directives, major blueprint refactor

### PR DESCRIPTION
**Pull request by dtaylor113**
Prior to this PR, the Blueprint Editor (blueprint name, toolbox tabs/items, blueprint details dlg., etc.), and the Blueprint Canvas (add, remove, select, and tag nodes. connection mode, canvas key bindings, etc.), were in the same large HTML file, controlled via two controllers.

This PR creates new `<blueprint-editor>` and `<blueprint-canvas>` directives in order to better differentiate and separate editor .vs canvas business logic and code.  

This was also done in preparation for blueprint editor and canvas unit testing which will follow this PR.

This PR also contains a simple unit test for the blueprint editor state controller.

### CODE REVIEW

File: client/app/components/blueprints/blueprint-editor/blueprint-editor.html [hunk 1]
```diff
@@ -0,0 +1,116 @@
+<span ng-class="{'wait-cursor': vm.loading}">
+  <span class="blueprint-designer-header">
+    <ol class="breadcrumb">
+        <li>
+            <a ui-sref="^"> <span class="fa fa-angle-double-left"/>&nbsp;{{'Back to Blueprints'|translate}}</a>
+        </li>
+        <li class="active"><strong translate>Blueprint Editor: </strong>
```
Comment (chriskacerguis) [inline client/app/components/blueprints/blueprint-editor/blueprint-editor.html]: Are we doing "translate" on a strong?  Do we not want to use the "{{'Back to Blueprints'|translate}}" format?

File: client/app/components/blueprints/blueprint-editor/blueprint-editor.html [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+          <button id="toggleToolbox" class="btn btn-primary" ng-class="{'disabled': vm.inConnectingMode}" type="button" ng-click="vm.showToolbox()"
+                  data-toggle="tooltip" data-placement="bottom" title="Add Service Item To Canvas" translate>Add Item
+          </button>
+          <button id="duplicateItem" class="btn" type="button" ng-class="{'disabled': !vm.onlyOneTtemSelected() || vm.inConnectingMode}"
+                  ng-click="vm.duplicateSelectedItem()" data-toggle="tooltip" data-placement="bottom"
+                  title="Duplicate Item (Ctrl-D)" translate>Duplicate
+          </button>
+          <button id="removeItems" class="btn" type="button" ng-class="{'disabled': !vm.itemsSelected() || vm.inConnectingMode}"
+                  ng-click="vm.removeSelectedItemsFromCanvas()" data-toggle="tooltip" data-placement="bottom"
+                  title="Remove Items From Canvas" translate>Remove
```
Comment (chriskacerguis) [inline client/app/components/blueprints/blueprint-editor/blueprint-editor.html]: Do we need to translate the title?

File: client/app/components/blueprints/blueprint-editor/blueprint-editor.html [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+  <div class="blueprint-designer-container">
+
+      <div class="toolbar">
+          <button id="toggleToolbox" class="btn btn-primary" ng-class="{'disabled': vm.inConnectingMode}" type="button" ng-click="vm.showToolbox()"
+                  data-toggle="tooltip" data-placement="bottom" title="Add Service Item To Canvas" translate>Add Item
+          </button>
+          <button id="duplicateItem" class="btn" type="button" ng-class="{'disabled': !vm.onlyOneTtemSelected() || vm.inConnectingMode}"
+                  ng-click="vm.duplicateSelectedItem()" data-toggle="tooltip" data-placement="bottom"
+                  title="Duplicate Item (Ctrl-D)" translate>Duplicate
```
Comment (chriskacerguis) [inline client/app/components/blueprints/blueprint-editor/blueprint-editor.html]: Do we need to translate the title?

File: client/app/components/blueprints/blueprint-editor/blueprint-editor.html [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+            </button>
+        </li>
+    </ol>
+  </span>
+
+  <div class="blueprint-designer-container">
+
+      <div class="toolbar">
+          <button id="toggleToolbox" class="btn btn-primary" ng-class="{'disabled': vm.inConnectingMode}" type="button" ng-click="vm.showToolbox()"
+                  data-toggle="tooltip" data-placement="bottom" title="Add Service Item To Canvas" translate>Add Item
```
Comment (chriskacerguis) [inline client/app/components/blueprints/blueprint-editor/blueprint-editor.html]: Do we need to translate the title?

Comment (dtaylor113) [inline client/app/components/blueprints/blueprint-editor/blueprint-editor.html]: Yes, will do.

Comment (dtaylor113) [inline client/app/components/blueprints/blueprint-editor/blueprint-editor.html]: Fixed

File: client/app/components/blueprints/blueprint-editor/blueprint-editor.html (hunk 2 above)
Comment (dtaylor113) [inline client/app/components/blueprints/blueprint-editor/blueprint-editor.html]: Fixed

File: client/app/components/blueprints/blueprint-editor/blueprint-editor.html (hunk 1 above)
Comment (dtaylor113) [inline client/app/components/blueprints/blueprint-editor/blueprint-editor.html]: Found about 12 other examples of `<strong translate...>` in SUI code base :-)

Comment (chriskacerguis) [inline client/app/components/blueprints/blueprint-editor/blueprint-editor.html]: Yes, I just want to make sure that we are following what @mzazrivec 's established pattern is.

**Comment by chriskacerguis** [comment]
Going to merge, haven't gotten any response, so I'll just go with @dtaylor113 thoughts about it already exists in the codebase.

**Pull request merged**